app-text/asciidoc-9.0.0_rc1: version bump

marcecj opened the pull request:
Hi,

I recently noticed that AsciiDoc upstream tagged a release candidate for the upcoming asciidoc-py3 version 9.0.0, so I thought it would be prudent to go ahead and package it.  While at it I added a missing test dependency.  I also removed test support from the current stable version (8.6.10_p20181016), since up to now the reference output was not included in the source distribution, meaning running the test suite would compare output generated by the current version with itself, which is not particularly informative.  This changed with 9.0.0rc1, so that gets to keep the test feature.

(Note that I'm not sure whether I need to do a revbump for any of the post-bump commits, but I think that they're OK since users probably shouldn't be installing stuff with FEATURES="test".)

Neither pkgcheck nor repoman report any errors.  The test suite passes for both Python 3.6 and PyPy3 (I didn't test Python 3.7).

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -0,0 +1,92 @@
+	python_fix_shebang "${ED}"/usr/bin/*.py
+
+	readme.gentoo_create_doc
+	dodoc BUGS.txt CHANGELOG.txt README.asciidoc docbook-xsl/asciidoc-docbook-xsl.txt \
```
Since you're already calling `default` in this phase, you should add a global `DOCS` array somewhere before `src_` phases containing these files.

marcecj replied:
OK

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -0,0 +1,92 @@
+		dev-util/source-highlight
+		media-gfx/graphviz
+		media-gfx/imagemagick
+		media-sound/lilypond
```
This seems weird...

marcecj replied:
What do you mean?  Are you referring to the difference in dependencies between USE="test" and otherwise?  If so: While asciidoc supports three alternatives for source highlighting, the reference output in the tests were created with source-highlight, so the tests specifically require that.

juippis replied:
`media-sound/lilypond` but I didnt really dig into upstream what does it do.

marcecj replied:
Oh, I wasn't sure myself (I've heard of lilypond, but never used it myself), see the file at /usr/share/doc/asciidoc-8.6.10_p20181016/examples/music-filter.txt.  Looks like another candidate for optfeature.

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -0,0 +1,92 @@
+	dev-libs/libxslt
+	dev-libs/libxml2
+	${PYTHON_DEPS}
+	graphviz? ( media-gfx/graphviz )
```
Should probably be an `optfeature` since it doesn't have any function in this ebuild, 
https://projects.gentoo.org/qa/policy-guide/dependencies.html#optional-runtime-dependencies

marcecj replied:
Yeah, I could change that, but it's always been like that in asciidoc.  But I guess a version bump is as good an opportunity as any.  I'll do it in a clean-up commit though, that will consolidate some of your other suggestions (the ones that aren't directly relevant to the version bump itself).

marcecj replied:
One question: will removing the USE flag cause a problem for the asciidoc `package.use.mask` entry in `profiles/arch/riscv/package.use.mask`?

juippis replied:
It will be fine, there's still versions left in tree. But I wasn't aware of such mask, it's good to know so it may be removed once obsolete.

marcecj replied:
OK

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -0,0 +1,92 @@
+	>=app-text/docbook-xsl-stylesheets-1.75
+	dev-libs/libxslt
+	dev-libs/libxml2
+	${PYTHON_DEPS}
```
Please put this on top or to the bottom of this list, not in the middle.

marcecj replied:
OK

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -0,0 +1,92 @@
+	app-text/docbook-xml-dtd:4.5
+	>=app-text/docbook-xsl-stylesheets-1.75
+	dev-libs/libxslt
+	dev-libs/libxml2
```
Remember to add slot, `:2`

marcecj replied:
OK

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -35,6 +35,7 @@ RDEPEND="
 DEPEND="
 	test? (
 		app-text/dvipng
+		app-text/dvisvgm
```
Please squash this into the original version bump.

marcecj replied:
I disagree, it's not directly relevant to the version bump (the dependency was always missing).

juippis commented on app-text/asciidoc/asciidoc-8.6.10_p20181016.ebuild:
```diff
@@ -15,8 +15,7 @@ KEYWORDS="alpha amd64 arm arm64 hppa ia64 m68k ~mips ppc ppc64 ~riscv s390 sh sp
 LICENSE="GPL-2"
 SLOT="0"
-IUSE="examples graphviz highlight test"
-RESTRICT="!test? ( test )"
```
You can just add `RESTRICT="test"` and put a short comment on top of it explaining why it's disabled. I'd much prefer that over to removing these pieces of information what was/is needed for test phase.

marcecj replied:
What pieces of information are you referring to that would be missing? The same bits of information are in the new 9.0.0_rc1 ebuild.  But whichever you prefer, I can also just not touch the test functionality, but then I have to add the dvisvgm dependency like I did for 9.0.0_rc1.

juippis replied:
I don't think touching this ebuild at all is very relevant anymore, doubt you will get bug reports for it. It's good if they are fixed in a later version.

marcecj replied:
OK

juippis commented on app-text/asciidoc/asciidoc-9.0.0_rc1.ebuild:
```diff
@@ -66,7 +66,6 @@ src_configure() {
 
 src_test() {
 	local -x ASCIIDOC_PY=asciidoc.py
-	"${EPYTHON}" tests/test${PN}.py update || die
```
Please squash into the original version bump.

marcecj replied:
OK

marcecj commented:
@juippis : Thanks for the review; I addressed most of your comments locally and will wait until we're agreed on the rest before pushing any updates.

WRT me removing test support in the current stable version: I just felt that it was misleading to support `FEATURES="test"` when the test suite doesn't really do what you expect.  Theoretically it does still test asciidoc in a limited way (e.g., none of the code paths crash), but it doesn't actually verify whether the outputs are correct.  I'm starting to think it would be best to just leave the test support in place and add a comment along the lines of this paragraph.  I would then prepend a separate commit that adds the `app-text/dvisvgm` dependency to (instead of adding it to the 9.0.0_rc1 ebuild later).

What do you think?

marcecj commented:
I just rebased these patches on current master, mainly due to commit ebeb21006bcc87339ba74b3b05ffe4770108a470 "app-text/asciidoc: Switch to PYTHON_MULTI_USEDEP API".

marcecj commented:
Thanks for merging (and for fixing those nits).  You seem to have forgotten a `Closes` tag, though.
